### DISCUSSION TITLE
View items from a collection that are tied to a user

### DISCUSSION THREAD

**Discussion by kaiquefeng** [directus/directus#11707] [Archive]
Hi people,

I created a collection with a field of [user_created] that for the uuid of the user who created a new item in the collection, but I wanted to make it so that when he went to view the items in the collection, he could only view those that have the same [user_created] field to its own uuid, both in the interface and via the API.

I used a filter on the front to consume the API, however this is not safe

![image](https://user-images.githubusercontent.com/56850504/154708396-3a769886-271e-4482-97f0-06d508a2bf73.png)

basically I'm doing this on the front, but I want to do this validation directly in directus for greater security

**Comment by rijkvanzanten** [comment]
Goeiemorgen! Ik spreek helaas geen Portugees, dus kan je niet verder helpen. Dat gezegd hebbende, misschien kan @joselcvarela je even te woord staan.

**Reply by kaiquefeng** [reply]
Hi people,

I created a collection with a field of [user_created] that for the uuid of the user who created a new item in the collection, but I wanted to make it so that when he went to view the items in the collection, he could only view those that have the same [user_created] field to its own uuid, both in the interface and via the API.

I used a filter on the front to consume the API, however this is not safe

![image](https://user-images.githubusercontent.com/56850504/154709627-7ffef199-cfdd-4d67-baea-b5b76719c34d.png)

basically I'm doing this on the front, but I want to do this validation directly in directus for greater security

**Reply by kaiquefeng** [reply]
@rijkvanzanten can you help me with this?

**Reply by rijkvanzanten** [reply]
Now I can! 😄 

You can set the permissions for your `invoices` collection to 

```
{
  user_created: {
    _eq: "$CURRENT_USER"
  }
}
```

to only allow users to retrieve their own invoices 👍🏻

**Reply by kaiquefeng** [reply]
where do I add this validation in the collection?

**Reply by kaiquefeng** [reply]
insert this validation in a specific collection?

**Reply by joselcvarela** [reply]
https://user-images.githubusercontent.com/14039341/154711563-8e40f6bf-6154-4240-b174-0652bf9cd2e3.mov

**Reply by kaiquefeng** [reply]
thanks!! helped me a lot
